### PR TITLE
Phoenix 1 3

### PR DESCRIPTION
**Pull request by ericsullivan**

### CODE REVIEW

File: lib/curator/extension.ex [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      def authenticated_plugs(), do: nil
+      def unauthenticated_routes(), do: nil
+      def authenticated_routes(), do: nil
+
+      # def before_sign_in(_, _), do: :ok
+      # def after_sign_in(conn, _, _), do: conn
+      # def after_failed_sign_in(conn, _, _), do: conn
+      # def after_extension(conn, _, _), do: conn
+
+      defoverridable [
```
Comment (aditya7iyengar) [inline lib/curator/extension.ex]: This is so cool! :blue_heart: :heart_eyes:

File: lib/curator/router.ex [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    web_module = Curator.Config.web_module()
+    module_quotes = get_module_quotes(modules, :unauthenticated_routes)
+
+    quote do
+      scope "/auth", unquote(web_module) do
+        # get "/session/new", AuthController, :new
+        # post "/session", AuthController, :create
+        delete "/session", SessionController, :delete
+
+        unquote(module_quotes)
```
Comment (aditya7iyengar) [inline lib/curator/router.ex]: :clap:

File: lib/curator/ueberauth/ueberauth.ex [hunk 3]
```diff
@@ -0,0 +1,11 @@
+defmodule Curator.Ueberauth do
+  use Curator.Extension
+
+  def unauthenticated_routes do
+    quote do
+      get "/:provider", UeberauthController, :request
```
Comment (aditya7iyengar) [inline lib/curator/ueberauth/ueberauth.ex]: I think we might have to make this more configurable.. Their auth controller might have a different name. Maybe `curator_ueberauth` config could have a key for `auth_controller` and `action`?

File: lib/curator/router.ex (hunk 2 above)
Comment (ericsullivan) [reply lib/curator/router.ex]: Took far too long to figure out...

**Pull request merged**